libs/portal-integration-angular/src/lib/services/portal-dialog.service.ts

```diff
@@ -1,8 +1,138 @@
-import { Injectable } from '@angular/core'
-import { DialogService } from 'primeng/dynamicdialog'
-
-/**
- * This is a bit of a hack. For some reason, providing MessageService does not work correctly across federated modules
- */
-@Injectable({ providedIn: 'root' })
-export class PortalDialogService extends DialogService {}
+import { EventEmitter, Injectable, Type } from '@angular/core'
+import { TranslateService } from '@ngx-translate/core'
+import { Observable } from 'rxjs'
+import { DialogService, DynamicDialogConfig, DynamicDialogRef } from 'primeng/dynamicdialog'
+
+import { ButtonDialogComponent } from '../core/components/button-dialog/button-dialog.component'
+import { ButtonDialogButtonDetails, ButtonDialogData } from '../model/button-dialog'
+import { DialogHostComponent } from '../core/components/button-dialog/dialog-host/dialog-host.component'
+
+type TranslationKeyWithParameters = { key: string; parameters: Record<string, unknown> }
+type TranslationKey = string | TranslationKeyWithParameters
+type DialogMessage = { message: TranslationKey; icon: string }
+export interface DialogResult<T> {
+  dialogResult: T
+}
+export interface DialogPrimaryButtonDisabled {
+  primaryButtonEnabled: EventEmitter<boolean>
+}
+export interface DialogSecondaryButtonDisabled {
+  secondaryButtonEnabled: EventEmitter<boolean>
+}
+export interface DialogButtonClicked {
+  ocxDialogButtonClicked(state: DialogState<any>): Observable<boolean> | Promise<boolean> | boolean | undefined
+}
+
+type Component<T> = {
+  type: Type<any> | DialogResult<T>
+  inputs?: Record<string, unknown>
+}
+
+export type DialogState<T> = {
+  button: 'primary' | 'secondary'
+  result: T | undefined
+}
+
+@Injectable({ providedIn: 'any' })
+export class PortalDialogService {
+  constructor(private dialogService: DialogService, private translateService: TranslateService) {}
+
+  /**
+   * @deprecated
+   */
+  open(componentType: Type<any>, config: DynamicDialogConfig): DynamicDialogRef {
+    return this.dialogService.open(componentType, config)
+  }
+
+  openDialog<T = unknown>(
+    title: TranslationKey | null,
+    componentOrMessage: Component<T> | TranslationKey | DialogMessage,
+    primaryButtonTranslationKeyOrDetails: TranslationKey | ButtonDialogButtonDetails,
+    secondaryButtonTranslationKeyOrDetails?: TranslationKey | ButtonDialogButtonDetails,
+    showCloseButton: boolean = true
+  ): Observable<DialogState<T>> {
+    let dialogTitle = ''
+    const translateParams = this.prepareTitleForTranslation(title)
+    this.translateService.get(translateParams.key, translateParams.parameters).subscribe((translation: string) => {
```

Comment:
Please do not use subscribe to get the value out of an Observable. `dialogTitle` should stay an Observable till it is used in the view somewhere with `| async`

Reply:
dialogTitle property is used by `this.dialogService.open(ButtonDialogComponent, {
      header: dialogTitle,
      data: dynamicDialogDataConfig,
      closable: showCloseButton && secondaryButtonTranslationKeyOrDetails !== undefined,
    }).onClose`

`header` is expected to be type of string and we have no control on how it is being rendered. Is this still applicable here?